File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
self_number = [True] * 10001

for i in range(1, 10000):
    if i + sum(map(int, str(i))) < 10001:
        self_number[i + sum(map(int, str(i)))] = False

for i in range(1, 10001):
    if self_number[i]:
        print(i)

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.java
public class Main {
    public static void main(String[] args) {
        boolean[] A = new boolean[10001];
        for (int i = 0; i < 10001; i++) A[i] = true;
        
        for (int i = 1; i < 10000; i++) {
            String num = Integer.toString(i);
            
            int sum = i;
            for (int j = 0; j < num.length(); j++) sum += (num.charAt(j) - '0');
            
            if (sum < 10001) A[sum] = false;
        }
        
        for (int i = 1; i < 10001; i++) {
            if (A[i]) System.out.println(i);
        }
    }
}